relays/lib-substrate-relay/src/finality/mod.rs

```diff
@@ -51,14 +50,7 @@ pub(crate) const RECENT_FINALITY_PROOFS_LIMIT: usize = 4096;
 
 /// Substrate -> Substrate finality proofs synchronization pipeline.
 #[async_trait]
-pub trait SubstrateFinalitySyncPipeline: 'static + Clone + Debug + Send + Sync {
-	/// Headers of this chain are submitted to the `TargetChain`.
-	type SourceChain: Chain;
-	/// Headers of the `SourceChain` are submitted to this chain.
-	type TargetChain: ChainWithTransactions;
-
-	/// Finality engine.
-	type FinalityEngine: Engine<Self::SourceChain>;
+pub trait SubstrateFinalitySyncPipeline: SubstrateFinalityPipeline {
 	/// How submit finality proof call is built?
 	type SubmitFinalityProofCallBuilder: SubmitFinalityProofCallBuilder<Self>;
```

Comment:
why `SubmitFinalityProofCallBuilder` was not moved to `SubstrateFialityPipeline`?
I could see `SubstrateFinalityPipeline` as a abstract "holder of data" / "configuration" and `SubstrateFinalitySyncPipeline` adding some  behavior (also adds `[async_trait]`).


if we move `SubmitFinalityProofCallBuilder` to `SubstrateFinalityPipeline`,
then `SubstrateFinalitySyncPipeline` could be more like `GuardedSubstrateFinalityPipeline`

Reply:
bellow I see:

```
pub type SubstrateFinalityProof<P> = <<P as SubstrateFinalityPipeline>::FinalityEngine as Engine<
	<P as SubstrateFinalityPipeline>::SourceChain,
>>::FinalityProof;
```

so the proof is related to the `SubstrateFinalityPipeline`, so `SubmitFinalityProofCallBuilder` should be placed maybe there